src/pages/auth/login.tsx

```diff
@@ -54,7 +54,8 @@
         setIsError(true);
         return;
       }
-      router.push('/manage');
+      const redirectAfterLogin = router.query['redirectAfterLogin'] as string;
+      router.push(redirectAfterLogin ?? '/manage');
```

Comment:
This seems to be actually an issue, because you are able to redirect for example to `data:text/html,<script>alert('hi');</script>` to show the user an alert

Reply:
Yes, I dislike my current solution a lot. Could we simply use zod or regex to validate?

Reply:
@Meierschlumpf please-review again. I'll add a few unit tests to confirm the functionality in a few minutes.

Reply:
Has the issue been fixed? - Do we regex check it or something?

Reply:
https://github.com/ajnart/homarr/pull/1416/files#diff-22cefed7cb2125befb9a6a2df36a695c8ab59681e8aba3051866759ff26f3e4fR145-R149

Reply:
Oh my bad sorry